<a href="https://colab.research.google.com/github/SALRIGO/py28/blob/master/notebook/mid_exam_py28.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###파이썬활용 중간고사 풀이

##문제1
- #다음 식의 연산 결과는?

In [0]:
a = 5  #a에 5라는 값 지정
b = 3  #b에 3이라는 값지정
a**b   #a^b = 5^3  

- ▲ 15
- ◆ 243 
- ● 125  (o)

    5^3 = 5×5×5 = 125
- ■ 2

## 문제2
- 다음 문자열 a에 대하여 슬라이싱을 한 결과는?

In [0]:
a = "Life is too short, You need Python!"  # a 에 "Life is too short, You need Python!" 라는 문자열 지정
a[-16:-8]                                  # a의 뒤에서 16번째 문자('Y')부터 뒤에서 8번째 문자(d' 'P) 전까지 출력
#'You need'

'You need'

- ▲ 'You need' (o)

    a[-16:-8]
- ◆ 'you need Python'

    a[19:-1]
- ● 'need Python'

    a[23:-1]
- ■ 'short, You'

    a[12:-13]

## 문제3
- 다음 list a의 인덱싱 결과는?

In [54]:
a = [1,2,(3,4),['a','b',['my','Life','Good']]] #리스트 a 생성
a[3][-1][0]                                     
                                                #a리스트 안에있는 4번째 리스트(['a','b',['my','Life','Good']])
                                                #내의 뒤에서 1번째에 있는 리스트(['my','Life','Good']) 
                                                #내의 1번째에 있는 'my'                                              

'my'

- ▲ 'a'

    a[3][0]
- ◆ 'My'  (o)

    a[3][-1][0]
- ● 'Life'

    a[3][2][1]
- ■ 'Good'

    a[3][2][2]

## 문제4
- 다음 tuple t1의 슬라이싱 결과는?

In [0]:

t1 = (0,1,"s","t","u")           #t1 에 튜플 생성
t1[2:-2]                         #튜플 t1에서 3번째 자리("s")부터 뒤에서 2번째 자리("t")전 까지 슬라이싱
                                 #튜플은 값이 하나만 출력될수 없기 떄문에 ('s',)가 출력됩니다. 하나만 출력할 경우 str

str

- ▲ ('s','t')

    t1[2:-1]    type:tuple
- ◆ ('s','t',)

    출력불가
- ● ('s')

    t1[2]       type:str
- ■ ('s',)    (o)

    t1[2:-2]    type:tuple

## 문제5
- 다음 bool 값 출력 중 정확한 것은?

In [4]:
bool([0,1,2]), bool([0]),bool((0)),bool((0,))
#bool([0,1,2])는 [0,1,2]이라는 값이 리스트로 있어서 True
#bool([0])는 [0]이라는 값이 리스트로 있어서 True
#bool((0))는 값이 0인 int 이기 때문에 그로인해 (0 = False) False     (1=True)
#bool(0,)는 (0,)이라는 튜플이기때문에 True

(True, True, False, True)

- ▲ (True, True, True, True)
- ◆ (False, False, False, False)
- ● (True, True, False, True)  (o)
- ■ (True, False, False, True)

## 문제6
- 다음 조건문을 1줄 코드로 변경하시오.

In [15]:
score = 80                    #score에 값80 지정
if score >= 60:               #score의 값이 60 보다 크거나 같으면 message에 문자열 "success"을 대입
  message = "success"          
else:                         #그렇지 않을경우 message에 문자열 "failure"을 대입
  message = "failure"
#아래의 답중에 위의 연산과 같은 경우는 ◆이기때문에 정답이라 판단하였습니다.
print(message)

success


- ▲ message = "failure" if score>=60 else "success"

    failure
- ◆ message = "success" if score>=60 else "failure" (o)

    success
- ● message = "failure" if score>=60 then "success"

    SyntaxError: invalid syntax
- ■ message = "success" if score>=60 then "failure"

    SyntaxError: invalid syntax

##문제7
- 다음 코드의 실행 결과는?

In [32]:
games = ["A","B","C"]             #game에 ["A","B","C"]라는 리스트 생성

for cnt in range(len(games)):     #for 반복구문 반복할때마다 cnt에 0부터 3-1 의 값을 한번씩 집어넣음 
  print(cnt, games[cnt], end=',')  #현재 cnt와 games리스트의 cnt번쨰의 문자 출력후 ,를 찍고 가로로 출력

0 A,1 B,2 C,

- ▲ 0 A,1 B,2 C, (o)
- ◆ 0 A,1 B,2 C
- ● 1 A,2 B,3 C,
- ■ 1 A,2 B,3 C

##문제8
- 다음의 결과가 출력되지 않는 것은? 'I ate 10 apples, so I was sick for three days'

In [38]:
# code start
number = 10
day = "three"

"I ate {0} apples. so I was sick for {1} days.".format(10,day)         #format함수를 이용하여 출력값: 10, three(day에 저장된 문자열)
f"I ate {number} apples. so I was sick for {day} days."                #f문자열을 이용한 포매팅 출력값: 10(number에저장됨), three(day에 저장됨)
f"I ate {number} apples. so I was sick for {'three'} days."            #f문자열을 이용한 포매팅 출력값: 10(number에저장됨), three
"I ate {0} apples. so I was sick for {1} ays.".format(number=10, day)  #SyntaxError: positional argument follows keyword argument (number=10은형식x)

SyntaxError: ignored

- ▲ "I ate {0} apples. so I was sick for {1} days.".format(10,day)

    'I ate 10 apples. so I was sick for three days.'
- ◆ f"I ate {number} apples. so I was sick for {day} days."

    'I ate 10 apples. so I was sick for three days.'
- ● f"I ate {number} apples. so I was sick for {'three'} days."

    'I ate 10 apples. so I was sick for three days.'
- ■ "I ate {0} apples. so I was sick for {1} ays.".format(number=10, day) (o)

    SyntaxError: positional argument follows keyword argument

##문제9
- 리스트 내포를 사용해 1부터 1000이하의 자연수 중 5의 배수의 합을 구하는 한 줄 코드를 작성하시오.

In [0]:
5 + 10 + 15 + 20 + 25 + .....
... + 955 + 960 + 965 + 970 +
975 + ... + 995 + 1000 = ?

In [43]:
sum([num for num in range(1,1001) if num % 5==0]) #1에서 1001전까지의 숫자를 한번씩num에 집어넣고 만약 집어넣은 값이 5로 나누었을때
                                                  #나머지가 0으로 떨어질 경우 sum함수를 이용하여 값을 중복하여 더함

100500

- ▲ sum([num for num in range(1,1000) if num % 5==0])

    range함수는 (a,b)일떄 a숫자 부터 b숫자의 전까지 수를 반환 하기때문에
    해당연산의 경우 숫자가 **999**까지 반환되기 떄문에 1000이 누락되어       
    더해지지 않아 99500라는 값이 나오게된다.
- ◆ len([num for num in range(1,1000) if num % 5==0])

    len 함수의 경우 num안에들은 갯수를 세주는 함수이기 떄문에 1에서  
    999까지의 숫자중 5의 배수의 갯수를 세주어서 **199**라는 값이 출력된다.
- ● sum([num for num in range(1,1001) if num % 5==0]) (o)

100500
- ■ len([num for num in range(1,1001) if num % 5==0])

    1에서 1000까지의 숫자중 5의 배수의 갯수를 세기때문에 **200**이라는 값이 
    출력된다.

##문제10
- 4개 과목의 평균 성적을 구하시오. [a],[b]에 적합한 코드는?

In [0]:
#문제
subjects = ['python','english','math','science']
scores = [88, 77, 66, 95]
total = 0

for subject, score in [a].______(subjects, scores):
  total += score
  print(subject, score)

print("평균 점수 = (0)".format([b]._______))  

In [53]:
#▲
subjects = ['python','english','math','science']       #subjects에 리스트 지정
scores = [88, 77, 66, 95]                              #scores에 리스트 지정
total = 0
                                                       #밑의 답처럼 zip함수의 사용방법으로 enumerate 함수를 사용할경우 오류가 발생한다.
for subject, score in enumerate(subjects, scores):     #오류를 방지하려면  subjects를 제외할경우 작동하게된다
  total += score                                       #score의값들을 더하여 total에 집어넣음
  print(subject, score)                                #subject와 score 값 출력

print("평균 점수 = {0}".format(score/len(subjects)))  #문자열 포메팅을 이용하여 score를 4로 나눈 값이 출력된다.

TypeError: ignored

In [48]:
#◆
subjects = ['python','english','math','science']        #subjects에 리스트 지정
scores = [88, 77, 66, 95]                               #scores에 리스트 지정
total = 0

for subject, score in zip(subjects, scores):           #zip함수를 이용하여subject에는 subjects의 값을 score에는 scores의 값을 하나씩 넣는다.
  total += score                                       #score의값들을 더하여 total에 집어넣음
  print(subject, score)                                #subject와 score 값 출력

print("평균 점수 = {0}".format(len(subjects)))         #문자열 포메팅을 이용하여 subjects리스트의 수(4)를 센값을 출력

python 88
english 77
math 66
science 95
평균 점수 = 4


In [49]:
#●
subjects = ['python','english','math','science']       #subjects에 리스트 지정
scores = [88, 77, 66, 95]                              #scores에 리스트 지정
total = 0
                                                       #밑의 답처럼 zip함수의 사용방법으로 enumerate 함수를 사용할경우 오류가 발생한다.
for subject, score in enumerate(subjects, scores):     #오류를 방지하려면  subjects를 제외할경우 작동하게된다
  total += score                                       #score의값들을 더하여 total에 집어넣음
  print(subject, score)                                #subject와 score 값 출력

print("평균 점수 = {0}".format(total/len(subjects)))   #total에다 subjects리스트의 갯수(4)를 값을 출력한다.

TypeError: ignored

In [50]:
#■
subjects = ['python','english','math','science']          #subjects에 리스트 지정
scores = [88, 77, 66, 95]                                 #scores에 리스트 지정
total = 0

for subject, score in zip(subjects, scores):              #zip함수를 이용하여subject에는 subjects의 값을 score에는 scores의 값을 하나씩 넣는다.
  total += score                                          #score의값들을 더하여 total에 집어넣음
  print(subject, score)

print("평균 점수 = {0}".format(total/len(subjects)))      #total에다 subjects리스트의 갯수(4)를 값을 출력한다.

python 88
english 77
math 66
science 95
평균 점수 = 81.5



- ▲ enumerate, scores/len(subjects)

    TypeError: 'list' object cannot be interpreted as an integer
- ◆ zip,scores/len(subjects)

    평균 점수 = 4
- ● enumerate, total/len(subjects)

    TypeError: 'list' object cannot be interpreted as an integer
- ■ zip, total/len(subjects)  (o)

    평균 점수 = 81.5